api/views/admin_auth.py

```diff
@@ -62,6 +81,118 @@ def register():
         return make_response(jsonify(response_object), 400)
 
 
+@admin_auth.route('/confirmation/<confirmation_id>', methods=['GET'])
+def confirm_registration(confirmation_id):
+    """ Endpoint that verifies user's signup process through a confirmation ID from email """
+
+    def is_valid_uuid(val):
+        try:
+            uuid.UUID(str(val))
+            return True
+        except ValueError:
+            return False
+
+    if not is_valid_uuid(confirmation_id):
+        response_object = {
+            "status": False,
+            "message": "Invalid confirmation parameter."
+        }
+
+        return make_response(jsonify(response_object), 401)
+
+    user = users.update_one(
+        {"emailConfirmationId": confirmation_id},
+        {
+            "$set": {"emailConfirmed": True},
+            "$unset": {"emailConfirmationId": ""}
+        },
+        upsert=False
+    )
+
+    if not user.acknowledged:
+        response_object = {
+            "status": False,
+            "message": "Invalid ID. Please register a new user or resend the verification email."
+        }
+        return make_response(jsonify(response_object), 401)
+
+    if user.modified_count == 0:
```

Comment:
More like a question: why 0 not 1? since ur deleting the fields (emailConfirmed and emailConfirmationId).

Reply:
user.modified_count is a parameter that tells if an object has been updated or not — in the case that the user becomes acknowledged (exists) but not parameter becomes updated is something we also need to consider as well.